ossec_security/assets/logs/ossec-security.yaml

```diff
@@ -0,0 +1,708 @@
+              [preauth]"
+            - "May  2 12:14:01 ub20 sshd[1573405]: Disconnecting authenticating
+              user root 10.10.10.10 port 1000: Too many authentication failures
+              [preauth]"
+            - "May 29 18:43:12 ub20 sshd[254289]: PAM 2 more authentication
+              failures; logname= uid=0 euid=0 tty=ssh ruser= rhost=10.10.10.10
+              user=dummy"
+          grok:
+            supportRules: |-
+              DATETIME (%{date("MMM  d HH:mm:ss"):}|%{date("MMM d HH:mm:ss"):})
+              SKIPHEADER %{regex("[^:]*"):}
+            matchRules: >-
+              multiple_authentication_fail_parse_rule %{DATETIME:}
+              %{SKIPHEADER:}: Disconnecting authenticating user
+              %{notSpace:usr.name} %{ip:src_ip} port
+              %{integer:network.client.port}\: Too many authentication
+              failures%{data}
+
+
+              authentication_fail_parse_rule %{DATETIME:} %{SKIPHEADER:}: PAM 2 more authentication failures; %{data::keyvalue("=","/")}
+        - type: attribute-remapper
+          name: Map `user` to `usr.name`
+          enabled: true
+          sources:
+            - user
+          sourceType: attribute
+          target: usr.name
+          targetType: attribute
+          preserveSource: false
+          overrideOnConflict: false
+        - type: attribute-remapper
+          name: Map `uid` to `usr.id`
+          enabled: true
+          sources:
+            - uid
+          sourceType: attribute
+          target: usr.id
+          targetType: attribute
+          preserveSource: false
+          overrideOnConflict: false
+        - type: attribute-remapper
+          name: Map `rhost` to `network.client.ip`
+          enabled: true
+          sources:
+            - rhost
+          sourceType: attribute
+          target: network.client.ip
+          targetType: attribute
+          preserveSource: false
+          overrideOnConflict: false
+    - type: pipeline
+      name: Parsing of DPKG Logs
+      enabled: true
+      filter:
+        query: "@rule_id:[2900 TO 2903] "
+      processors:
+        - type: grok-parser
+          name: Parse DPKG logs
+          enabled: true
+          source: log
+          samples:
+            - 2024-05-27 12:08:24 status installed systemd:amd64
+              245.4-4ubuntu3.23
+            - 2008-04-01 16:10:43 remove nmap 4.11-1 4.11-1
+            - 2024-05-29 06:14:50 remove curl 4.0.3-0.3 4.0.3-0.3
+            - 2008-04-01 14:39:49 install screen <none> 4.0.3-0.3
+            - 2008-04-01 16:10:43 status half-installed nmap 4.11-1
+          grok:
+            supportRules: ""
+            matchRules: >-
+              dpkg_package_parse_rule_with_status %{date("yyyy-MM-dd HH:mm:ss")}
+              status %{notSpace:operation} %{data:package_name}
+
+
+              dpkg_package_parse_rule %{date("yyyy-MM-dd HH:mm:ss")} %{notSpace:operation} %{notSpace:package_name}%{data}
```

Comment:
same for here, you can replace ${data} by %{notSpace}( %{notSpace})?

Reply:
Updated as per suggestion